engine/src/main/java/org/hibernate/search/engine/impl/nullencoding/NullMarkerCodec.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.impl.nullencoding;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.search.Query;
+import org.hibernate.search.bridge.LuceneOptions;
+
+/**
+ * Contract to manage coding and decoding (querying) of null values.
+ * Null values can be stored as special markers or skipping the field
+ * altogether.
+ * When encoding a marker the index field type shall match the other fields,
+ * as index types often need to be consistent across the whole index.
+ * Also the strategy to find a match for such a null-encoded value might vary.
+ *
+ * @author Sanne Grinovero
+ */
+public interface NullMarkerCodec {
+
+	/**
+	 * This is mostly a requirement for integration with other old-style
+	 * contracts which expect a strongly String based strategy.
+	 * @return a string representation of the null-marker, or null if no marker is used.
+	 */
+	String nullRepresentedAsString();
+
+	/**
+	 * Store the null marker in the Document.
+	 * Some implementations might not do anything
+	 * @param fieldName the
+	 * @param document
+	 * @param luceneOptions
+	 */
+	void encodeNullValue(String fieldName, Document document, LuceneOptions luceneOptions);
```

Comment:
good point, but it's an internal contract and we can changes this whenever we want. I did try other paths but the FieldBridge contract of today didn't leave me much freedom .. I'd need more invasive changes.